### ISSUE TITLE
Bug: Lots of open tabs lead to UI regression

### ISSUE THREAD

**Issue by Overload119** [notaapp/issues#32]
![ScreenGrab 2020-12-14 at 11 19 13](https://user-images.githubusercontent.com/1012072/102106218-4bf0ec80-3dfe-11eb-9dbe-3a1ea5487840.png)

**Comment by led** [comment]
### DESCRIPTION:
![Screenshot 2021-06-24 at 08 14 34](https://user-images.githubusercontent.com/395440/123219426-cc3a8d80-d4c4-11eb-9e49-517187890aac.png)

When very many tabs are open, the tab label is reduced to a single character. This is not enough information to make the tabs useful or an effective way to work with your files.

### EXPECTED:
Tab labels to be readable in order to be able to select from them effectively.

### POSSIBLE SOLUTIONS:
- don't truncate the labels so severely
- make the tabs a pane with horizontal scrolling
- overflow tabs into a menu on right-hand side
- order the tabs by most recently used, and _fully_ display the labels on the most recent \<arbitrary number\> of tabs, and use another approach for the remaining tabs

**Comment by erusev** [comment]
> When very many tabs are open, the tab label is reduced to a single character. This is not enough information to make the tabs useful or an effective way to work with your files.

I agree. I'll add this to the short-term roadmap. Thanks for the solutions 🙏
